lib/agents_exporter.rb

```diff
@@ -32,14 +33,26 @@ def links
     agent_ids = agents.map(&:id)
 
     contained_links = agents.map.with_index do |agent, index|
-      agent.links_as_source.where(:receiver_id => agent_ids).map do |link|
-        { :source => index, :receiver => agent_ids.index(link.receiver_id) }
+      agent.links_as_source.where(receiver_id: agent_ids).map do |link|
+        { source: index, receiver: agent_ids.index(link.receiver_id) }
       end
     end
 
     contained_links.flatten.compact
   end
 
+  def control_links
+    agent_ids = agents.map(&:id)
```

Comment:
Oh, gotcha.  `options[:agents]` is already a cached array, so I think this is okay.